src/azul/service/responseobjects/buffer.py

```diff
@@ -0,0 +1,55 @@
+from logging import getLogger
+from io import BytesIO
+from typing import Callable
+
+logger = getLogger(__name__)
+
+
+class FlushableBuffer(BytesIO):
+    """
+    A buffer that flushes the output to a callback function (``inbetween_callback``),
+    either if there exists the size of the fragment is large enough
+    (over ``chunk_size``) on flush or when the buffer is closed.
+
+    :param chunk_size: The minimum size of buffer (byte length)
+    :param inbetween_callback: The callback function to receive flushed output
+    """
+
+    def __init__(self, chunk_size: int, inbetween_callback: Callable):
+        super(FlushableBuffer, self).__init__()
+        self.__chunk_size = chunk_size
+        self.__inbetween_callback = inbetween_callback
+        self.__remaining_size = 0
+
+    def write(self, b: bytes):
+        super().write(b)
+        byte_count = len(b)
+        self.__remaining_size += byte_count
+        self.__clean_up()
+
+    def close(self):
+        if self.__remaining_size > 0:
+            logger.warning(f'Clearing the remaining buffer (approx. {self.__remaining_size} B)')
+            self.__inbetween_callback(self.getvalue())
+            self.__remaining_size = 0
+            # As the buffer is closed, the pointer doesn't need to be reset.
+        super().close()
+
+    def __clean_up(self):
+        if self.__remaining_size < self.__chunk_size:
+            return
+        value = self.getvalue()
```

Comment:
Consider getbuffer() to avoid the copy.  But only if the change is easy.

Reply:
Somehow, invoking `getbuffer()` causes the object to be unresizable.

```BufferError: Existing exports of data: object cannot be re-sized```

This will prevent the buffer from truncating/resetting itself.

Reply:
See docs on `getbuffer()`. You might have to `del` the return value before calling `truncate()`

Reply:
I tried that. It works. But I also looked into a different way. The just updated code relies on the stream position `BytesIO.seek` and the memory efficiency should be the same. Can you check it out and let me know what you think?